### PR TITLE
Bugfixes, removeAll() functionality

### PR DESCRIPTION
**Pull request by teddywilson**

### CODE REVIEW

File: core/src/main/scala/blobstore/fs/FileStore.scala [hunk 1]
```diff
@@ -65,7 +65,11 @@ case class FileStore[F[_]](fsroot: NioPath)(implicit F: Effect[F]) extends Store
 
   override def move(src: Path, dst: Path): F[Unit] = F.delay(Files.move(src, dst)).void
 
-  override def copy(src: Path, dst: Path): F[Unit] = F.delay(Files.copy(src, dst)).void
+  override def copy(src: Path, dst: Path): F[Unit] = {
+    F.delay(Files.createDirectories(_toNioPath(dst).getParent)).as(true).map { _ =>
+      Files.copy(src, dst)
```
Comment (stew) [inline core/src/main/scala/blobstore/fs/FileStore.scala]: .as(true) coerces the values inside F to be true, and then the map call just throws these away, right? we could skip the call to .as

Comment (stew) [reply core/src/main/scala/blobstore/fs/FileStore.scala]: I don't think we should assume that F is well behaved if an exception is thrown by the function passed to map, so I think that this should be a `.flatMap(_ => F.delay(Files.copy...))`

Comment (stew) [reply core/src/main/scala/blobstore/fs/FileStore.scala]: and `fa.flatMap(_ => foo)` can be replaced by `fa *> foo`

Comment (stew) [reply core/src/main/scala/blobstore/fs/FileStore.scala]: or both operations could happen inside the same F.delay

Comment (rolandomanrique) [reply core/src/main/scala/blobstore/fs/FileStore.scala]: @teddywilson I would go with Stew's last option... both operations inside a single F.delay

**Comment by teddywilson** [comment]
should be ready for another pass:
cc: @rolandomanrique @stew

**Pull request merged**